Documentation/Tasks/Index.rst

```diff
@@ -0,0 +1,169 @@
+* **releasePath**: The path to the release directory in work (typically referenced by *next*)
+* **sharedPath**: The path to the shared directory for all releases
+* **currentPath**: The path that points to the *current* release
+* **previousPath**: The path that points to the *previous* release
+
+Add task to the deployment flow
+-------------------------------
+
+So we have seen how to create custom tasks in different ways. In the following we will see how we add these tasks to the deployment flow::
+
+   <?php
+
+   ...
+
+   $application = new \\TYPO3\\Surf\\Application\\TYPO3\\CMS();
+   $deployment->addApplication($application);
+   $workflow = $deployment->getWorkflow();
+
+   $workflow->defineTask('CopyEnvFileTask', \\TYPO3\\Surf\\Task\\ShellTask::class, [
+      'command' => [
+         "cp {sharedPath}/.env {releasePath}/.env",
+         "cd {releasePath}",
+      ]
+   ]);
+
+   $deployment->onInitialize(function() use ($workflow, $application) {
+      $workflow->afterStage('transfer', 'CopyEnvFileTask', $application);
+   });
+
+This will execute the new task after the stage transfer only for the application referenced by $application.
+
+Besides specifying the execution point via a stage, you can also give an existing task as an anchor and specify the task execution with **afterTask** or **beforeTask**::
+
+   <?php
+
+   ....
+
+   $workflow->beforeTask(CreatePackageStatesTask::class,
+        [
+            'CopyEnvFileTask'
+        ]
+    );
+
+
+The following table shows all the methods to manipulate the tasks in the deployment flow (part of the abstract Workflow class):
+
+====================== ================================= ===================================================================================
+Method                 Arguments                         Description
+====================== ================================= ===================================================================================
+defineTask             $taskName, $taskType, ($options)  Defines a new task with name $taskName based on $taskType with custom options.
+addTask                $tasks, $stage, ($application)    Add one or more tasks to the workflow that should run in the given stage.
+removeTask             $taskName                         Removes the task with the given name from all stages and applications.
+afterTask              $taskName, $tasks, ($application) Adds one or more tasks that should run *after* the given task name.
+beforeTask             $taskName, $tasks, ($application) Adds one or more tasks that should run *before* the given task name.
+====================== ================================= ===================================================================================
+
+Options for Task
+----------------
+
+In order to customize options of existing tasks you can do it the following ways::
+
+   <?php
+
+   use TYPO3\\Surf\\Task\\Transfer\\RsyncTask;
```

Comment:
add hint, how to get the options from the tasks, as we don't have a complete documentation of them already